README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

[![pub package](https://img.shields.io/pub/v/color_log.svg?logo=dart&logoColor=00b9fc)](https://pub.dev/packages/color_log)
[![Last Commits](https://img.shields.io/github/last-commit/aditechdev/color_log?logo=git&logoColor=white)](https://github.com/aditechdev/color_log/commits/main)
[![Pull Requests](https://img.shields.io/github/issues-pr/aditechdev/color_log?logo=github&logoColor=white)](https://github.com/aditechdev/color_log/pulls)
[![Code size](https://img.shields.io/github/languages/code-size/aditechdev/color_log?logo=github&logoColor=white)](https://github.com/aditechdev/color_log)
[![License](https://img.shields.io/github/license/aditechdev/color_log?logo=open-source-initiative&logoColor=green)](https://github.com/aditechdev/color_log/blob/main/LICENSE)

<!-- [![CI](https://img.shields.io/github/actions/workflow/status/Bungeefan/logger/dart.yml?branch=master&logo=github-actions&logoColor=white)](https://github.com/Bungeefan/logger/actions) -->

ColorLog is a Flutter package that provides colorful and emoji-filled logs, making them more noticeable and easier to read.

By using this package, developers can customize their log output, avoiding the usual messy blue text.

With different emojis included, logs become more expressive and eye-catching. 

Please show support for the project by 

liking ❤️ and starring ⭐⭐⭐⭐⭐ the repo, and don't forget to share ✈️ it with other developers.
                                                                           
                                            I ❤️ love emoji



## Features

This package has 8 varities of log
- info : This will help the developer to get info
- warning: To see the warning
- error: To check the error, add it in the api call, catch
- debug: To use while debuging
- checkSuccess: To check boolean value, which will give print the log in true and false
- custom: This will give you freedom to use color, emoji and log type of your choice, 

## Getting started

Search for the package color log in pub.dev

## Usage

```dart
    class Home extends StatelessWidget {
  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            Button(
              onTap: () {
                clog.info("This is debug log");
              },
              text: "info",
              color: Colors.cyan,
            ),
            Button(
              onTap: () {
                clog.warning("This is warning log");
              },
              text: "warning",
              color: Colors.yellow,
            ),
            Button(
              onTap: () {
                clog.error("This is error log");
              },
              text: "error",
              color: Colors.red,
            ),
            Button(
              onTap: () {
                try {
                  throw Exception;
                } catch (e) {
                  clog.error(e.toString());
                  rethrow;
                }
              },
              text: "error with stacktrace",
              color: Colors.red,
            ),
            Button(
              onTap: () {
                clog.debug("This is debug log");
              },
              text: "debug",
              color: Colors.yellow,
            ),
            Button(
              onTap: () {
                clog.checkSuccess(1 == 1, "This is debug log");
              },
              text: "checkSuccess true",
              color: Colors.green,
            ),
            Button(
              onTap: () {
                clog.checkSuccess(1 != 1, "This is debug log");
              },
              text: "checkSuccess false",
              color: Colors.red,
            ),
            Button(
              onTap: () {
                clog.todo("This is TODO");
              },
              text: "TODO",
              color: Colors.pink,
            ),
            Button(
              onTap: () {
                clog.custom(
                    ansiColor: "\x1b[38;5;205m",
                    emoji: "🤓",
                    logType: "custom",
                    message: "This is custom");
              },
              text: "custom",
              color: Colors.pink,
            ),
          ],
        ),
      ),
    );
  }
}

class Button extends StatelessWidget {
  const Button({super.key, this.onTap, this.text, this.color});
  final VoidCallback? onTap;
  final String? text;
  final Color? color;

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 50,
      color: color,
      margin: const EdgeInsets.all(16),
      width: MediaQuery.of(context).size.width,
      child: ElevatedButton(
        // style: MaterialStateProperty.all(value),
        style: ElevatedButton.styleFrom(
          backgroundColor: color,
        ),

        onPressed: onTap,
        onHover: (c) {},
        child: Text(
          text!,
        ),
      ),
    );
  }
}



```



```dart
String temp = "colorLog"
clog.info($temp);
clog.warning($temp);
clog.error($temp);
clog.debug($temp);
clog.checkSuccess(1 == 1, "TO Verify 1==1");
clog.checkSuccess(1 == 2, "TO Verify 1==2");
clog.custom(message: "This is custom log", emoji: "🍃" , ansiColor: "Any ANSI color", logType: "custom");


```

## Additional information

To see more detail go to github

created by [Aditechdev](https://github.com/aditechdev).


CHANGELOG.md
## 0.0.1

* Release color log with emoji

## 0.0.2

* Added description and small change for package
## 0.0.3

* Added description and small change for package
## 0.0.4

* Simplify the description
## 0.0.5

* Small change
## 0.0.6

* Small change


test/color_log_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:color_log/color_log.dart';

void main() {
  test('adds one to input values', () {
    // final calculator = Calculator();
    // expect(calculator.addOne(2), 3);
    // expect(calculator.addOne(-7), -6);
    // expect(calculator.addOne(0), 1);
  });
}


lib/color_log.dart
library color_log;

import 'package:flutter/foundation.dart';

enum LogLevel { info, warning, error, debug, wtf, success, fail, todo }

/// A Calculator.
class ColorLog {
  // final String _reset = '\x1B[0m';
  final String _red = '\x1B[31m';
  final String _green = '\x1B[32m';
  final String _yellow = '\x1B[33m';
  // final String _bold = '\x1B[1m';
  final String _cyan = '\x1b[36m';
  // final String _purple = '\x1B[35m';
  final String _pink = "\x1b[38;5;205m";

  /// To get information about something
  ///
  /// The log color will be cyan
  void info(String message) {
    printTheMessage(
        color: _cyan,
        emoji: "⛅",
        message: message,
        logType: LogLevel.info.name);
  }

  /// To log sometype of warning
  ///
  /// The log color will be yellow
  void warning(String message) {
    printTheMessage(
        color: _yellow,
        emoji: "🐢",
        message: message,
        logType: LogLevel.warning.name);
  }

  /// To log error in red color always rethrow or throw after using `error` log to get the stacktrace in the console
  ///
  /// The log color will be Red
  void error(String message) {
    printTheMessage(
        color: _red,
        emoji: "⛔",
        message: message,
        logType: LogLevel.error.name);
  }

  /// For temporary debuging the app, remove this after debig
  ///
  /// The log color will be yellow
  void debug(String message) {
    printTheMessage(
        color: _yellow,
        emoji: "🚧",
        message: message,
        logType: LogLevel.debug.name);
  }

  // ///WTF is to track
  // ///
  // /// The log color will be yellow
  // void wtf(String message) {
  //   printTheMessage(
  //       color: _red, emoji: "⛔", message: message, logType: LogLevel.wtf.name);
  // }

  ///To log boolean check operation
  ///
  ///IF true the color will be green else color will be red
  checkSuccess(bool isSuccess, String message) {
    if (isSuccess) {
      printTheMessage(
          color: _green, emoji: "✅", message: message, logType: "TRUE");
    } else {
      printTheMessage(
          color: _red, emoji: "❌", message: message, logType: "FALSE");
    }
  }

  ///To log todo check operation
  ///
  ///This will always remind your todo in log
  todo(String message) {
    printTheMessage(
        color: _pink,
        emoji: "📝",
        message: message,
        logType: LogLevel.todo.name);
  }

  ///This is a custom log
  ///
  ///Need to add all the custom properties
  void custom(
      {required String message,
      required String emoji,
      required String ansiColor,
      required String logType}) {
    printTheMessage(
        color: ansiColor, emoji: emoji, message: message, logType: logType);
  }
}

/// Print in color and emoji
printTheMessage(
    {String emoji = "✅",
    String message = "",
    String color = '\x1B[0m',
    String logType = "info"}) {
  if (kDebugMode) {
    String bold = '\x1B[1m';
    String normal = '\x1B[0m';
    String reset = '\x1B[0m';

    print('$color$bold ---------------------------------------------');

    print(
        '$emoji $color$bold[${logType.toUpperCase()}]:$normal$color $message$reset');
    // print(
    //     '$color$bold --------------------------------------------------------------');
  }
}

ColorLog clog = ColorLog();


example/lib/main.dart
import 'package:color_log/color_log.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            Button(
              onTap: () {
                clog.info("This is debug log");
              },
              text: "info",
              color: Colors.cyan,
            ),
            Button(
              onTap: () {
                clog.warning("This is warning log");
              },
              text: "warning",
              color: Colors.yellow,
            ),
            Button(
              onTap: () {
                clog.error("This is error log");
              },
              text: "error",
              color: Colors.red,
            ),
            Button(
              onTap: () {
                try {
                  throw Exception;
                } catch (e) {
                  clog.error(e.toString());
                  rethrow;
                }
              },
              text: "error with stacktrace",
              color: Colors.red,
            ),
            Button(
              onTap: () {
                clog.debug("This is debug log");
              },
              text: "debug",
              color: Colors.yellow,
            ),
            Button(
              onTap: () {
                clog.checkSuccess(1 == 1, "This is debug log");
              },
              text: "checkSuccess true",
              color: Colors.green,
            ),
            Button(
              onTap: () {
                clog.checkSuccess(1 != 1, "This is debug log");
              },
              text: "checkSuccess false",
              color: Colors.red,
            ),
            Button(
              onTap: () {
                clog.todo("This is TODO");
              },
              text: "TODO",
              color: Colors.pink,
            ),
            Button(
              onTap: () {
                clog.custom(
                    ansiColor: "\x1b[38;5;205m",
                    emoji: "🤓",
                    logType: "custom",
                    message: "This is custom");
              },
              text: "custom",
              color: Colors.pink,
            ),
          ],
        ),
      ),
    );
  }
}

class Button extends StatelessWidget {
  const Button({super.key, this.onTap, this.text, this.color});
  final VoidCallback? onTap;
  final String? text;
  final Color? color;

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 50,
      color: color,
      margin: const EdgeInsets.all(16),
      width: MediaQuery.of(context).size.width,
      child: ElevatedButton(
        // style: MaterialStateProperty.all(value),
        style: ElevatedButton.styleFrom(
          backgroundColor: color,
        ),

        onPressed: onTap,
        onHover: (c) {},
        child: Text(
          text!,
        ),
      ),
    );
  }
}
